code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm

```diff
@@ -55,7 +55,7 @@
 
 
 /obj/machinery/icemachine/proc/validexchange(reag)
-	if(reag == "sprinkles" | reag == "cola" | reag == "kahlua" | reag == "dr_gibb" | reag == "vodka" | reag == "space_up" | reag == "rum" | reag == "spacemountainwind" | reag == "gin" | reag == "cream" | reag == "water")
+	if(reag == "sprinkles" | reag == "cola" | reag == "kahlua" | reag == "dr_gibb" | reag == "vodka" | reag == "space_up" | reag == "rum" | reag == "spacemountainwind" | reag == "gin" | reag == "cream" | reag == "vanilla" | reag == "whiskey" | reag == "brownstar")
```

Comment:
do you think you could change this logic to check list membership while you're at it? This is a little nasty.

Something like 
```
var/list/static/invalid_reagents = list(...)
if(reag in invalid_reagents)
    return

Reply:
to be honest, partially because i'm brain-fried, i'm not _entirely_ certain what this proc is supposed to be doing. If i'm understanding what its supposed to do, is block certain reagents from being transferred into the machine? If that's it, I'd say it's basically removable.